src/lib/chat/matrix-client.ts

```diff
@@ -189,6 +189,10 @@ export class MatrixClient implements IChatClient {
       if (event.type === 'm.room.message') {
         this.events.receiveNewMessage(event.room_id, (await mapMatrixMessage(event, this.matrix)) as any);
       }
+
+      if (event.type === 'm.room.create') {
+        this.roomCreated(event);
```

Comment:
Looks good. I've been thinking that maybe the user presence data should be event based too. Going to look at re-working it a little